### PR TITLE
Add state and reducer for results table

### PR DESCRIPTION
**Pull request by kimberlythegeek**
blocked by #148

### CODE REVIEW

File: src/components/Results/ResultsTable.tsx [hunk 1]
```diff
@@ -55,33 +36,31 @@ function ResultsTable() {
...
+              newRuns,
+            } = result;
+            return (
+              <TableRow key={index}>
+                <TableCell>{platformName}</TableCell>
+                <TableCell>{testName}</TableCell>
+                <TableCell>{baseValue}</TableCell>
+                <TableCell>{newValue}</TableCell>
+                <TableCell>{delta}%</TableCell>
+                <TableCell>{confidenceText}</TableCell>
```
Comment (eng-esther) [inline src/components/Results/ResultsTable.tsx:57]: [suggestion] do you want to use a map to iterativelly generate the cells?

Comment (kimberlythegeek) [reply src/components/Results/ResultsTable.tsx:57]: I considered this, but was unsure what we will be doing in the future, so mapping may not work. example, for delta, when we have `is_improvement` and `is_regression`, we'll likely use conditional rendering to display either a check icon or x icon.

File: src/common/store.ts [hunk 2]
```diff
@@ -2,12 +2,14 @@ import { configureStore } from '@reduxjs/toolkit';
...
 
 import checkedRevisions from '../reducers/CheckedRevisions';
+import results from '../reducers/ResultsSlice';
 import search from '../reducers/SearchSlice';
 import selectedRevisions from '../reducers/SelectedRevisions';
 
 const reducer = combineReducers({
-  search,
   checkedRevisions,
+  results,
```
Comment (eng-esther) [inline src/common/store.ts:11]: [question] would we have other types of results?
maybe this should be renamed to compareResults ? 🤷‍♀️

**Review by eng-esther — APPROVED** [review]

File: src/reducers/CompareResultsSlice.ts [hunk 3]
```diff
@@ -0,0 +1,64 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+// TODO: replace with thunk that accesses endpoint
+// import { fetchCompareResults } from '../thunks/fetchCompareResults';
+import type { CompareResultsState } from '../types/state';
+
+const initialState: CompareResultsState = [
```
Comment (julienw) [inline src/reducers/CompareResultsSlice.ts:7]: (optional) A comment stating that they are test values could be useful

File: src/types/state.ts [hunk 4]
```diff
@@ -39,6 +52,8 @@ export type SelectedRevisionsState = {
   revisions: Revision[];
 };
 
+export type CompareResultsState = CompareResultsItem[];
```
Comment (julienw) [inline src/types/state.ts:55]: Don't you need to add `compareResults` below to the `State` too?

Looking at https://github.com/mozilla/perfcompare/pull/149/files#diff-c1cf636d61418b4439ce697bd7f4249145469dacd55453c8832c045f1a99f5cbR9-R14 it seems that some other properties are missing too. Is that expected that TypeScript doesn't error about that?

Comment (kimberlythegeek) [reply src/types/state.ts:55]: I originally was using this `State` type, but at some point added `RootState` that infers the type: https://github.com/mozilla/perfcompare/blob/master/src/common/store.ts#L21

I will file an issue to remove `State` and replace with `RootState`

my guess on why it isn't erroring is that the only place `State` is used, it is not accesing the missing state types: https://github.com/mozilla/perfcompare/pull/149/files#diff-f4aeacfb95826ff4c36316d81101147c633ca4a49655531c49c6dff7d4767970R33

**Pull request merged**